src/async/python/async.py

```diff
@@ -0,0 +1,121 @@
+"""Zaber Motion Library advanced async usage example for Python."""
+import asyncio
+from typing import AsyncIterator, List
+
+from zaber_motion import Units, UnitTable
+from zaber_motion.ascii import Connection
+
+# See ../README.md for more information.
+
+# Edit these constants to match your setup.
+PORT = "COM4"
+X_DEVICE_ADDRESS = 1
+X_AXIS_NUMBER = 1
+Y_DEVICE_ADDRESS = 1
+Y_AXIS_NUMBER = 2
+
+X_GRID_POINTS = 4
+Y_GRID_POINTS = 4
+GRID_SPACING = 5
+GRID_SPACING_UNITS = UnitTable.get_unit("mm")
+
+
+async def main() -> None:
+    """Demonstrate async iteration and Zaber Motion Library usage."""
+    # Connections are disposable because they hold system resources (ie a serial port or network
+    # connection). In an asynchronous program we recommend the "async with" statement to
+    # automatically close and dispose the Connection when finished with it.
+    #
+    # If you cannot encapsulate all Zaber device use in a single code block like this, you should
+    # make sure you manually close and dispose the Connection instance when your program is
+    # finished talking to devices.
+    async with Connection.open_serial_port_async(PORT) as connection:
+        # Enabling alerts speeds up detection of the end of device movement, but may cause problems
+        # for non-Zaber software communicating with the devices because it leaves them in a state
+        # where they can generate spontaneous messages. It is recommended if you are only using
+        # Zaber software (including the Zaber Motion Library).
+        await connection.enable_alerts_async()
+
+        # There is no async GetDevice because it just instantiates a device object
+        # without communicating with anything.
+        x_device = connection.get_device(X_DEVICE_ADDRESS)
+
+        # Everywhere you await a Zaber async function call, you have the option to do some other
+        # processing in parallel. Instead of using "await" immediately, you can assign the
+        # function's return value to a variable and then await it later after doing something else.
+        # You can also use asyncio.gather(...) to wait for multiple async operations to complete.
+        identify_task = x_device.identify_async()
+        # Could do something else here.
+        await identify_task  # Block until the identify_async() call completes.
```

Comment:
I don't think python will actually even start start the identification until this line. You would need to have
```
identify_task = asyncio.run(x_device.identify_async())
```
on the line above for it to start executing on its own.

Reply:
You're right. That does not meet my expectations. Changed it a bit.